dbms/src/Common/Exception.h

```diff
@@ -87,7 +90,8 @@ void tryLogCurrentException(Poco::Logger * logger, const std::string & start_of_
   * check_embedded_stacktrace - if DB::Exception has embedded stacktrace then
   *  only this stack trace will be printed.
   */
-std::string getCurrentExceptionMessage(bool with_stacktrace, bool check_embedded_stacktrace = false);
+std::string getCurrentExceptionMessage(bool with_stacktrace, bool check_embedded_stacktrace = false,
```

Comment:
Maybe always with extra info? What are the downsides?

Reply:
`getExtraExceptionInfo(...)` may catch an exception (not only if it's not Linux) while trying to collect extra info. In this case it appends the following message to extra info:
`msg += "\nCannot print extra info: " + getCurrentExceptionMessage(false, false, false);`
And `with_extra_info` must be false in this case to avoid infinite recursion.